### DISCUSSION TITLE
Resize partition of hard disk

### DISCUSSION THREAD

**Discussion by macamba** [bucanero/apollo-ps3#112] [Ideas]
Hi all,

I'm not sure if that makes sense suggest in this project but well I'll explain my problem
I bought 2TB ssd to install as internal memory in PS3 but I made mistake to not check maximun capability support in PS3 (maximum 1.5TB).
This ssd is from supplier Crucial and it doesnt let set max LBA by software, this is would be a good solution but not for my case.
I tried to clone usisng the tool HDD from Debug Setting of PS3 but at the end of the process PS3 frozen and not work
And tried to clone original HDD to SSD using application "Lazesoft Disk Image & Clone Home" from PC, this way PS3 detect ssd 2TB but it doesnt show all free space in this
So the question is, there is some posibility to add feature in Apollo Tool to resize manually partition ssd?

Thanks and regards

![PS3 size ssd](https://user-images.githubusercontent.com/37476319/230186248-bddfa715-ac07-43ee-9081-d5700810432b.jpg)

**Comment by Masamune3210** [comment]
No, the maximum partition size of MBR is 2TB. The ps3 will never be able to read a drive above that size, even if you put in a larger drive, the max size the partition will ever be is 2TB, which ends up with a roughly 1.85GB user partition, as shown in your pic

**Reply by Masamune3210** [reply]
The picture shows that you already have a 2TB partition, the reason it doesn't show all 2TB in the Free Space listing is that some of the drive is taken up by other things that the ps3 needs to use the drive for

**Reply by macamba** [reply]
Hi Masamune3210,

As I cloned the original hdd of 80GB to the new ssd 2TB sector by sector then PS3 shows the exactly partition of 80GB, so it makes me think about the issue is in size partition

Regards

**Reply by Masamune3210** [reply]
If your issue is that you cloned the drive and cant extend the partition out (I have bad news for you involving this), then why does your picture have the full 2TB?

Either way, due to the way the ps3 handles the hard drive encryption and its filesystem, there is no way to extend the size of a partition once it is made.

**Reply by macamba** [reply]
ok, thanks
I have some idea more but not sure if works,
I'm thinking to make a process with old HDD 3TB, changing Max LBA to 1.5TB, formating from PS3 and clone to ssd 2TB

Regards

**Reply by Masamune3210** [reply]
Why not just skip the cloning and just set the SSD up in the ps3 from the start? It should still be able to set up the drive, it just might not use the very last bit of it.

**Reply by macamba** [reply]
If I try to set or format the SSD 2TB directly in PS3, PS3 will be frozen
How you suggest do it exactly?

**Reply by Masamune3210** [reply]
If it freezes trying to set the drive up, it's likely also going to freeze when trying to read or write to the drive even if you clone it. It's the same underlying functions, the PS3 just can't handle drives that big

**Comment by bucanero** [comment]
Apollo focus is on save-games, so don't expect this app to solve any hard disk partition issue.

**Reply by macamba** [reply]
Ok, thanks anyway 😞
